tests/unit/igraph_maximum_matching.c

```diff
@@ -17,10 +17,133 @@
+        result = 2;
+    }
+    igraph_ring(&graph, 4, IGRAPH_UNDIRECTED, 0, 1);
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 3) {
+        printf("Length 3 path graph had %" IGRAPH_PRId " matching size.\n", matching_size);
+        result = 3;
+    }
+    return result;
+}
+
+int test_bipartite_graphs() {
+    int result = 0;
+    igraph_t graph;
+    igraph_vector_int_t matching;
+    igraph_integer_t matching_size;
+    //test two bipartite graphs
+    igraph_small(&graph, 9, IGRAPH_UNDIRECTED, 1,2, 2,3, 2,4, 2,5, 3,6, 4,7, 5,8, 6,9, -1);
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 4) {
+        printf("Bipartite graph 1 had %" IGRAPH_PRId " matching size, expected 4.\n", matching_size);
+        result = 1;
+    }
+    igraph_small(&graph, 6, IGRAPH_UNDIRECTED, 1,2, 1,3, 2,4, 3,4, 3,5, 4,6, 5,6, -1);
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 3) {
+        printf("Bipartite graph 2 had %" IGRAPH_PRId " matching size, expected 3.\n", matching_size);
+        result = 2;
+    }
+    return result;
+}
+
+int test_general_graphs() {
+    int result = 0;
+    igraph_t graph;
+    igraph_vector_int_t matching;
+    igraph_integer_t matching_size;
+    //test two non-bipartite graphs
+    igraph_small(&graph, 10, IGRAPH_UNDIRECTED, 1,2, 1,3, 2,9, 3,4, 3,5, 4,6, 5,7, 5,8, 7,8, 8,9, 8,10, -1);
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 5) {
+        printf("General graph 1 had %" IGRAPH_PRId " matching size, expected 5.\n", matching_size);
+        result = 1;
+    }
+    igraph_small(&graph, 6, IGRAPH_UNDIRECTED, 1,2, 1,3, 2,4, 3,5, 4,5, 5,6, -1);
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 3) {
+        printf("General graph 2 had %" IGRAPH_PRId " matching size, expected 3.\n", matching_size);
+        result = 2;
+    }
+    return result;
+}
 
-    //TODO: implement tests
-    return 0;
+int test_petersen_graph() {
+    int result = 0;
+    igraph_t graph;
+    igraph_vector_int_t matching;
+    igraph_integer_t matching_size;
+    //test petersen graph
+    igraph_famous(&graph, "Petersen");
+    igraph_maximum_matching(&graph, &matching_size, NULL, &matching, NULL);
+    if (matching_size != 5) {
+        printf("Petersen graph had %" IGRAPH_PRId " matching size, expected 5.\n", matching_size);
+        result = 1;
+    }
+    return result;
+}
+
+int main() {
+    int result = 0;
+    if (test_trivial_graphs()) {
+        result = 1;
+    }
+    if (test_path_graphs()) {
+        result = 1;
+    }
+    if (test_bipartite_graphs()) {
+        result = 1;
+    }
+    if (test_general_graphs()) {
+        result = 1;
+    }
+    if (test_petersen_graph()) {
+        result = 1;
+    }
+    return result;
```

Comment:
Just a heads up that we are moving away from using the return value for tests. You can use `IGRAPH_ASSERT()` or even `IGRAPH_FATAL()`. These have the advantage that when they fail, and ASan support is enabled, you will get a stack trace.